MagazineLayout/LayoutCore/Types/RowOffsetTracker.swift

```diff
@@ -0,0 +1,79 @@
+// Created by bryankeller on 5/23/19.
+// Copyright © 2019 Airbnb, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CoreGraphics
+
+/// Tracks offsets for rows using a Segment Tree for O(logn) lookups and updates.
```

Comment:
Comment clarifying that this is an implementation of a Segment Tree. There are tests added too to prevent regressions if anyone decides to mess with this 🙃

Reply:
#datastructures

Reply:
nit `O(log n)`